syntax = "proto3";

service FileService {
    rpc UploadFile(File) returns (FileResponse);
    rpc DeleteFile(FileName) returns (FileResponse);
    rpc RenameFile(RenameRequest) returns (FileResponse);
}

message File {
    string name = 1;
    bytes content = 2;
}

message FileName {
    string name = 1;
}

message RenameRequest {
    string old_name = 1;
    string new_name = 2;
}

message FileResponse {
    string message = 1;
}
